### PR TITLE
fix(select): fix select closing on chip removal

### PR DESCRIPTION
**Pull request by ismay**
This needs @HendrikThePendric's PR (https://github.com/dhis2/ui-core/pull/582) in order to work properly. But with that (and thus the correct z-indexes), it works as expected. Plus it gets rid of the global click listener, which is nice.

### CODE REVIEW

File: src/Select.js [hunk 1]
```diff
@@ -117,24 +115,16 @@ export class Select extends Component {
...
-
-        if (!isInsideClick) {
-            if (onBlur) {
-                onBlur({ selected }, e)
-            }
-
-            this.handleClose()
+        if (onBlur) {
+            onBlur({ selected }, e)
         }
```
Comment (Mohammer5) [inline src/Select.js]: This should be executed after `this.handleClose`, so any synchronous code called this way is not executed before the closing-logic has actually been executed.

File: src/Select.js [hunk 2]
```diff
@@ -205,43 +195,47 @@ export class Select extends Component {
...
-                        maxHeight={maxHeight}
-                        selectRef={this.selectRef}
-                        menuRef={this.menuRef}
-                        menuTop={menuTop}
-                        menuLeft={menuLeft}
-                        menuWidth={menuWidth}
-                    >
-                        {menu}
-                    </MenuWrapper>
+                    <ScreenCover onClick={this.onOutsideClick} transparent />
```
Comment (Mohammer5) [inline src/Select.js]: Why not rendering the Menu inside the ScreenCover, one less condition in the code? At least that's the intended usage of the ScreenCover.

I can see though how this works this way quite well.. so not a recommendation but rather a question

Comment (varl) [reply src/Select.js]: If you render it inside the `ScreenCover` the `ScreenCover` tries to align it in the middle of the screen by putting the `children` in a `div` with these styles:

```
div {
    position: absolute;
    top: 50%;
    left: 50%;
    width: auto;
    height: auto;
    transform: translate(-50%, -50%);
}
```

Comment (Mohammer5) [reply src/Select.js]: Oh, I forgot

File: src/Select.js (hunk 1 above)
Comment (varl) [reply src/Select.js]: I changed this in 129f14b.

**Pull request merged**